crates/chia-tools/src/bin/analyze-chain.rs

```diff
@@ -42,11 +42,14 @@ fn main() {
     let mut a = Allocator::new_limited(500_000_000);
     let allocator_checkpoint = a.checkpoint();
     let mut prev_timestamp = 0;
-    iterate_tx_blocks(
+    iterate_blocks(
         &args.file,
         args.start,
         Some(args.end),
         |height, block, block_refs| {
+            if block.transactions_generator.is_none() {
```

Comment:
What's the difference in all these, between `block.transactions_generator.is_none()` and `block.is_transaction_block()` aka `block.foliage.foliage_transaction_block_hash.is_some()` ?

Reply:
my understanding is that they are all either supposed to be `Some` or `None`.